charts/tenantfilesystem/templates/tenantfilesystem-composition.yaml

```diff
@@ -0,0 +1,854 @@
+            fromFieldPath: Required
+        - fromFieldPath: "spec.storageClass.uid"
+          toFieldPath: "spec.forProvider.manifest.parameters.uid"
+        - fromFieldPath: "spec.storageClass.gid"
+          toFieldPath: "spec.forProvider.manifest.parameters.gid"
+        - fromFieldPath: "spec.storageClass.gidRangeStart"
+          toFieldPath: "spec.forProvider.manifest.parameters.gidRangeStart"
+        - fromFieldPath: "spec.storageClass.gidRangeEnd"
+          toFieldPath: "spec.forProvider.manifest.parameters.gidRangeEnd"
+        - fromFieldPath: "spec.storageClass.basePath"
+          toFieldPath: "spec.forProvider.manifest.parameters.basePath"
+
+    - name: Role
+      base:
+        apiVersion: kubernetes.crossplane.io/v1alpha1
+        kind: Object
+        metadata:
+          labels:
+            crossplane/xrd-kind: xtenantfilesystem.wadtfy.bmwgroup.net
+            crossplane/xrd-helm_chart_version: {{ .Chart.Version }}
+        spec:
+          deletionPolicy: Delete
+          forProvider:
+            manifest:
+              apiVersion: rbac.authorization.k8s.io/v1
+              kind: Role
+              metadata:
+                namespace:
+              rules:
+              - apiGroups:
+                - ""
+                resources:
+                - secrets
+                verbs:
+                - get
+                - list
+          providerConfigRef:
+            name: provider-config-k8s
+      patches:
+        - type: PatchSet
+          patchSetName: "deletionPolicy"
+        - type: CombineFromComposite
+          combine:
+            variables:
+              - fromFieldPath: "spec.product"
+              - fromFieldPath: "metadata.labels[crossplane.io/claim-namespace]"
+              - fromFieldPath: "metadata.labels[crossplane.io/claim-name]"
+            strategy: string
+            string:
+              fmt: "efs-r-%s-%s-%s"
+          toFieldPath: "metadata.name"
+        - fromFieldPath: "metadata.labels[crossplane.io/claim-namespace]"
+          toFieldPath: "spec.forProvider.manifest.metadata.namespace"
+        - type: ToCompositeFieldPath
+          fromFieldPath: "metadata.name"
+          toFieldPath: "status.rbacRoleName"
+
+    - name: RoleBinding
+      base:
+        apiVersion: kubernetes.crossplane.io/v1alpha1
+        kind: Object
+        metadata:
+          labels:
+            crossplane/xrd-kind: xtenantfilesystem.wadtfy.bmwgroup.net
+            crossplane/xrd-helm_chart_version: {{ .Chart.Version }}
+        spec:
+          deletionPolicy: Delete
+          forProvider:
+            manifest:
+              apiVersion: rbac.authorization.k8s.io/v1
+              kind: RoleBinding
+              metadata:
+                namespace:
+              roleRef:
+                apiGroup: rbac.authorization.k8s.io
+                kind: Role
+                name:
+              subjects:
+              - kind: ServiceAccount
+                name: kube-system-sa
```

Comment:
Please put this in the global env-config EnvironmentConfig.  Whenever this name changes this component won't work anymore and hard-coding it at this place will break it